src/Modules/OrchardCore.Commerce/Services/PriceProvider.cs

```diff
@@ -27,8 +28,8 @@ public class PriceProvider : IPriceProvider
 
     public async Task<IEnumerable<ShoppingCartItem>> AddPricesAsync(IList<ShoppingCartItem> items)
     {
-        var skus = items.Select(item => item.ProductSku).Distinct().ToArray();
-        var skuProducts = (await _productService.GetProductsAsync(skus)).ToDictionary(productPart => productPart.Sku);
+        var skuProducts = await PriceProviderHelpers.GetSkuProductsAsync(items, _productService);
```

Comment:
If you transform that helper method into an extension on `IProductService`, this line will stay more idiomatic:
```suggestion
        var skuProducts = await _productService.GetSkuProductsAsync(items);
```